services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/extensions/dynamodb/mappingclient/operations/BatchGetItemOperation.java

```diff
@@ -15,65 +15,97 @@
 import software.amazon.awssdk.services.dynamodb.model.KeysAndAttributes;
 
-@SdkPublicApi
-public class ReadBatch<T> {
-    private final MappedTableResource<T> mappedTableResource;
-    private final Collection<BatchableReadOperation> readOperations;
+@SdkInternalApi
+public class BatchGetItemOperation
+    implements PaginatedDatabaseOperation<BatchGetItemRequest, BatchGetItemResponse, BatchGetResultPage> {
+
+    private final BatchGetItemEnhancedRequest request;
+
+    private BatchGetItemOperation(BatchGetItemEnhancedRequest request) {
+        this.request = request;
+    }
+
+    public static BatchGetItemOperation create(BatchGetItemEnhancedRequest request) {
+        return new BatchGetItemOperation(request);
+    }
+
+    @Override
+    public BatchGetItemRequest generateRequest(MapperExtension mapperExtension) {
+        Map<String, KeysAndAttributes> requestItems = new HashMap<>();
+        request.readBatches().forEach(readBatch -> addReadRequestsToMap(readBatch, requestItems));
 
-    private ReadBatch(MappedTableResource<T> mappedTableResource, Collection<BatchableReadOperation> readOperations) {
-        this.mappedTableResource = mappedTableResource;
-        this.readOperations = readOperations;
+        return BatchGetItemRequest.builder()
+                                  .requestItems(Collections.unmodifiableMap(requestItems))
+                                  .build();
     }
 
-    public static <T> ReadBatch<T> create(MappedTableResource<T> mappedTableResource,
-                                      Collection<BatchableReadOperation> readOperations) {
-        return new ReadBatch<>(mappedTableResource, readOperations);
+    @Override
+    public BatchGetResultPage transformResponse(BatchGetItemResponse response, MapperExtension mapperExtension) {
+        return BatchGetResultPage.builder().batchGetItemResponse(response).mapperExtension(mapperExtension).build();
     }
 
-    public static <T> ReadBatch<T> create(MappedTableResource<T> mappedTableResource,
-                                      BatchableReadOperation... readOperations) {
-        return new ReadBatch<>(mappedTableResource, Arrays.asList(readOperations));
+    @Override
+    public Function<BatchGetItemRequest, SdkIterable<BatchGetItemResponse>> serviceCall(DynamoDbClient dynamoDbClient) {
+        return dynamoDbClient::batchGetItemPaginator;
+    }
+
+    @Override
+    public Function<BatchGetItemRequest, SdkPublisher<BatchGetItemResponse>> asyncServiceCall(
+        DynamoDbAsyncClient dynamoDbAsyncClient) {
+
+        return dynamoDbAsyncClient::batchGetItemPaginator;
     }
 
-    void addReadRequestsToMap(Map<String, KeysAndAttributes> readRequestMap) {
-        KeysAndAttributes newKeysAndAttributes = generateKeysAndAttributes();
-        KeysAndAttributes existingKeysAndAttributes = readRequestMap.get(tableName());
+    private void addReadRequestsToMap(ReadBatch readBatch, Map<String, KeysAndAttributes> readRequestMap) {
```

Comment:
This can be rewritten to not suck anymore. It was only written this way due to a nuance in Java around raw type erasure, but now you've refactored it that no longer applies. We should just map it straight into requestItems.

Reply:
ok

Reply:
What did we ultimately decide to do about this?